gems/sorbet-runtime/lib/types/private/abstract/declare.rb

```diff
@@ -18,10 +18,6 @@ def self.declare_abstract(mod, type:)
     mod.extend(T::InterfaceWrapper::Helpers)
 
     if mod.is_a?(Class)
-      if mod < T::Struct
-        raise "#{mod.name} is a subclass of T::Struct and cannot be declared abstract"
-      end
-
```

Comment:
@azdavis can you confirm that you do not want these checks yet?

They were deleted in pay-server, but not deleted in Sorbet.

Reply:
Last I checked, pay-server breaks if this is not removed.